guide/10-mapping-and-visualization-alt/part7_utility_functions_for_the_map_widget.ipynb

```diff
@@ -0,0 +1,357 @@
+{
```

Comment:
<p>In this notebook, we will explore using utility functions to build interactivity. This can usually be split into two procedures: call back and display message. The <code>call back</code> function can be used to collect gis data on screen through digitization, while the <code>display message</code> function can be used to prompt users to perform a desired action. </p><p>To do this, we will need to setup an asynchronous <code>callback</code> function using the <code>on_click()</code> or <code>on_draw_end()</code> methods to create dynamic, interactive 'apps'. We will need to create a callback function, like <code>function_name(map_inst, geometry)</code>, with <code>map_inst</code> being the <code>MapView</code> instance and <code>geometry</code> being the <code>geometry</code> instance that the user is clicking. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1142/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>